src/dfx/src/lib/builders/motoko.rs

```diff
@@ -164,6 +164,7 @@ impl CanisterBuilder for MotokoBuilder {
                 .expect("Could not find canister ID."),
             wasm: WasmBuildOutput::File(motoko_info.get_output_wasm_path().to_path_buf()),
             idl: IdlBuildOutput::File(motoko_info.get_output_idl_path().to_path_buf()),
+            add_candid_service_metadata: false,
```

Comment:
For Motoko canister, if the did file is provided in `dfx.json`, we will want to override the generated metadata, ideally performing a subtyping check. But I'm not sure if `candid` field is present in for Motoko canisters at the moment?

Reply:
Yes, though that's beyond the scope of this PR.  For the time beyond, any `candid` field on dfx.json is not respected for Motoko canisters.